### PR TITLE
Added Tab Complete Functionality

### PR DESCRIPTION
**Pull request by verbeto**
Added functionality for tab complete.

### CODE REVIEW

**Comment by simon816** [comment]
Please follow the style guide. No tabs, 4 spaces per indentation level.

Also, remove 'Update MixinNetHandlerPlayServer.java' from the commit message, it doesn't mean anything.

**Comment by verbeto** [comment]
Fixed.

File: src/main/java/org/spongepowered/mod/mixin/server/MixinNetHandlerPlayServer.java [hunk 1]
```diff
@@ -79,4 +97,57 @@ public void sendCustomPayload(Object plugin, String channel, byte[] data) {
...
+            arraylist.add(value);
+        }
+        this.playerEntity.playerNetServerHandler.sendPacket(new S3APacketTabComplete((String[]) arraylist.toArray(new String[arraylist.size()])));
+
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private List<String> processTab(ICommandSender sender, String string) {
+
+        ArrayList arraylist = Lists.newArrayList();
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/mod/mixin/server/MixinNetHandlerPlayServer.java]: This doesn't need to be a rawtype.

**Comment by verbeto** [comment]
Warnings fixed also.

**Comment by kenzierocks** [comment]
You've got a few cast warnings.

**Comment by simon816** [comment]
This can be done neater with `@Inject` instead of copying the entire method.

You don't need to implement `INetHandlerPlayServer` as that is already an interface on the target class and you're missing an `@Overwrite` (Though should be `@Inject` ideally)

**Comment by simon816** [comment]
@verbeto Are you able to fix the above issues? Currently this will not work in an obfuscated environment.

**Comment by verbeto** [comment]
Apologies for the delay, the implementation has been rewritten and now uses the Overwrite annotation.    Unfortunately, as per the comment of the request, Inject cannot be used as more than one tab complete packet would result in only the final packet being displayed, and besides; implementing it this way gives us more control.

**Comment by simon816** [comment]
I think a better place for this would be in `MinecraftServer.func_180506_a` (getTabCompletions) or even `CommandHandler.getTabCompletionOptions`

**Comment by verbeto** [comment]
Well, I'll leave it like this- personally, I think separating vanilla code and sponge code as much as possible is best but, if need be, I can move the code into one of the methods you listed.

**Comment by zml2008** [comment]
This has been done with PR #215's refactoring

**Pull request closed without merging**